test/e2e/specs/data-json.js

```diff
@@ -0,0 +1,240 @@
+
+    expect(result.myJson).to.deep.include({
+      'letter'     : 'a',
+      'number'     : 11,
+      'decimals'   : [12.3, 43.28, 20],
+      'colours'    : ['red', 'green', 'blue'],
+      'description': 'It is an "Example".',
+      'timeMarks'  : {
+        'time'     : '12:18:29.000000',
+        'date'     : '2015-07-29',
+        'date_time': '2015-07-29 12:18:29.000000'
+      }
+    })
+  })
+
+  it('remove', async () => {
+    const jsonUpdateRemove = jsonUpdateBuilder.REMOVE()
+      .addArgument('$.timeMarks.date')
+      .addArgument('$.number')
+      .addArgument('$.colours[1]')
+      .create()
+
+    const newValues = { myJson: jsonUpdateRemove, objectId: savedObj.objectId }
+
+    const result = await TestTable.save(newValues)
+
+    expect(result.myJson).to.deep.include({
+      'letter'     : 'a',
+      'colours'    : ['red', 'blue'],
+      'decimals'   : [12.3, 43.28, 56.89],
+      'description': 'It is an "Example".',
+      'timeMarks'  : {
+        'time'     : '12:18:29.000000',
+        'date_time': '2015-07-29 12:18:29.000000'
+      }
+    })
+  })
+
+  it('array_append', async () => {
+    const jsonUpdateArrayAppend = jsonUpdateBuilder.ARRAY_APPEND()
+      .addArgument('$.decimals', 432.0)
+      .addArgument('$.colours', 'yellow')
+      .create()
+
+    const newValues = { myJson: jsonUpdateArrayAppend, objectId: savedObj.objectId }
+
+    const result = await TestTable.save(newValues)
+
+    expect(result.myJson).to.deep.include({
+        'letter': 'a',
+        'number': 10,
+        'decimals': [12.3, 43.28, 56.89, 432.0],
+        'colours': ['red', 'green', 'blue', 'yellow'],
+        'description': 'It is an "Example".',
+        'timeMarks': {
+          'time': '12:18:29.000000',
+          'date': '2015-07-29',
+          'date_time': '2015-07-29 12:18:29.000000'
+        }
+      })
+  })
+
+  it('array_insert', async () => {
+    const jsonUpdateArrayInsert = jsonUpdateBuilder.ARRAY_INSERT()
+      .addArgument('$.decimals[2]', 60)
+      .addArgument('$.colours[1]', 'cyan')
+      .create()
+
+    const newValues = { myJson: jsonUpdateArrayInsert, objectId: savedObj.objectId }
+
+    const result = await TestTable.save(newValues)
+
+    expect(result.myJson).to.deep.include({
+        'letter': 'a',
+        'number': 10,
+        'decimals': [12.3, 43.28, 60, 56.89],
+        'colours': ['red', 'cyan', 'green', 'blue'],
+        'description': 'It is an "Example".',
+        'timeMarks': {
+          'time': '12:18:29.000000',
+          'date': '2015-07-29',
+          'date_time': '2015-07-29 12:18:29.000000'
+        }
+      })
+  })
+
+  it('add no arguments and create', async () => {
+    let error
+
+    try {
+      jsonUpdateBuilder.INSERT().create()
+    } catch (e) {
+      error = e
+    }
+
+    expect(error.message).to.equal('You have to add at least one argument')
+  })
+
+  it('adding arguments without setting an operation', async () => {
```

Comment:
it seems like this is a Unit test, move it please to the Unit Tests directory

Reply:
+